xbmc/platform/win32/XTimeUtils.cpp

```diff
@@ -34,27 +32,30 @@ static_assert(offsetof(SystemTime, day) == offsetof(SYSTEMTIME, wDay));
 static_assert(offsetof(SystemTime, hour) == offsetof(SYSTEMTIME, wHour));
 static_assert(offsetof(SystemTime, minute) == offsetof(SYSTEMTIME, wMinute));
 static_assert(offsetof(SystemTime, second) == offsetof(SYSTEMTIME, wSecond));
+
+int64_t FileTimeToInt64(const FILETIME& ft)
+{
+  return static_cast<int64_t>((uint64_t(ft.dwHighDateTime) << 32) | ft.dwLowDateTime);
+}
+
 } // namespace
 
-uint32_t GetTimeZoneInformation(TimeZoneInformation* timeZoneInformation)
+std::tuple<bool, int64_t> GetTimezoneBias(const SystemTime& time)
 {
-  if (!timeZoneInformation)
-    return KODI_TIME_ZONE_ID_INVALID;
-
-  TIME_ZONE_INFORMATION tzi{};
-  uint32_t result = ::GetTimeZoneInformation(&tzi);
-  if (result == TIME_ZONE_ID_INVALID)
-    return KODI_TIME_ZONE_ID_INVALID;
-
-  timeZoneInformation->bias = tzi.Bias;
-  timeZoneInformation->daylightBias = tzi.DaylightBias;
-  timeZoneInformation->daylightDate = *reinterpret_cast<const SystemTime*>(&tzi.DaylightDate);
-  timeZoneInformation->daylightName = FromW(tzi.DaylightName);
-  timeZoneInformation->standardBias = tzi.StandardBias;
-  timeZoneInformation->standardDate = *reinterpret_cast<const SystemTime*>(&tzi.StandardDate);
-  timeZoneInformation->standardName = FromW(tzi.StandardName);
-
-  return result;
+  // Compute offset = local_filetime - utc_filetime
+  FILETIME utcFt{};
+  if (!::SystemTimeToFileTime(reinterpret_cast<const SYSTEMTIME*>(&time), &utcFt))
+    return {false, 0}; // error
+
+  FILETIME localFt;
+  if (!::FileTimeToLocalFileTime(&utcFt, &localFt))
+    return {false, 0}; // error
+
+  const int64_t u{FileTimeToInt64(utcFt)};
+  const int64_t l{FileTimeToInt64(localFt)};
```

Comment:
To avoid these intermediate variables and a conversion function, a union can be used:

The idea is have access to 2 x 32 bit members (FILETIME) or 64 bits using same variable:

```c++
std::tuple<bool, int64_t> GetTimezoneBias(const SystemTime& time)
{
  union FILETIME64
  {
    FILETIME ft;
    LONGLONG ft64;
  };

  // Compute offset = local_filetime - utc_filetime
  FILETIME64 utcFt{};
  if (!::SystemTimeToFileTime(reinterpret_cast<const SYSTEMTIME*>(&time), &utcFt.ft))
    return {false, 0}; // error

  FILETIME64 localFt{};
  if (!::FileTimeToLocalFileTime(&utcFt.ft, &localFt.ft))
    return {false, 0}; // error

  const int64_t gmtoff{(utcFt.ft64 - localFt.ft64) / 10000000LL / 60}; // convert 100-ns intervals -> minutes
  return {true, gmtoff};
}
```

Reply:
Thanks @thexai . I will change that.